src/ucp/proto/proto_single.c

```diff
@@ -62,10 +62,31 @@ ucp_proto_single_init_priv(const ucp_proto_single_init_params_t *params,
     return status;
 }
 
-ucs_status_t ucp_proto_single_init(const ucp_proto_single_init_params_t *params)
+static void
+ucp_proto_single_check_memory_access(ucp_proto_single_init_params_t *params)
```

Comment:
why is it only for proto single? i think it should be in proto_common

Reply:
Is there any sense to enable multi-fragment protocols on empty messages?

Reply:
some protocols (like put sw) have only multi fragment implementation

Reply:
Moved to proto_common ranges initialization logic.